app/waf/src/controllers/wafpolicy.controller.ts

```diff
@@ -109,8 +112,13 @@ export class WafpolicyController extends BaseController {
   ): Promise<Response> {
     return new Response(
       Wafpolicy,
-      await this.wafpolicyRepository.findById(id, undefined, {
-        tenantId: await this.tenantId,
+      await this.wafpolicyRepository.findById(id, {
+        where: {
+          and: [
+            {or: [{tenantId: await this.tenantId}, {public: true}]},
```

Comment:
Just thinking that `id` can identify anything. So there may be no need to `and` such 2 conditions. What do you think?

Reply:
I hope so, but it does not.

Reply:
why "does not" ?

Reply:
it becomes somethig like "{{or: [{xxx: xxx}, {xxx:xxx}]}, {"and":[{xxx:xxx}]}}"

Reply:
What I mean is if user passes an Id, we find the record with only 1 condition(id), no other condition.

Reply:
1. we want to get a public a wafpolicy whose tenantId is not the same as we used now.
2. if its without any other condition, we may get a wafpolicy which belongs to other tenant and is not public. or we can only get the wafpolicy which is in our own tenant (with the {tenantId: tenantId} option parameter).

Reply:
... Got it. Thanks.